### PR TITLE
feat(sparkplug): added Sparkplug B Protobuf Payload support and utility classes

### PR DESCRIPTION
**Pull request by marcellorinaldo**
This PR adds a Sparkplug topics utility and a Builder for encoding MQTT payload in Sparkplug B Protobuf format.

Dependencies:

- The `.proto` definition was taken from the official [Sparkplug v3.0.0 specification](https://sparkplug.eclipse.org/specification/) and is licensed under EPL 2.0.

**Related Issue:** N/A.

**Description of the solution adopted:**

- Added Builder for encoding metrics in Sparkplug B format and convert Kura TypedValues into compatible format
- Added topic utility
- Added tests

**Screenshots:** N/A.

**Manual Tests**: N/A.

**Any side note on the changes made:** N/A.

### CODE REVIEW

**Comment by marcellorinaldo** [comment]
Waiting for Eclipse approval on licensing since a third-party source code was included and modified. Will put on ready for review once OK.

File: kura/org.eclipse.kura.cloudconnection.sparkplug.mqtt.provider/pom.xml
Comment (pierantoniomerlino) [inline kura/org.eclipse.kura.cloudconnection.sparkplug.mqtt.provider/pom.xml]: Please remove comment.

File: kura/org.eclipse.kura.cloudconnection.sparkplug.mqtt.provider/src/main/java/org/eclipse/kura/cloudconnection/sparkplug/mqtt/message/SparkplugBProtobufPayloadBuilder.java
Comment (pierantoniomerlino) [inline kura/org.eclipse.kura.cloudconnection.sparkplug.mqtt.provider/src/main/java/org/eclipse/kura/cloudconnection/sparkplug/mqtt/message/SparkplugBProtobufPayloadBuilder.java]: Is it correct? If it is only for time value, maybe it is better to change the name of the method.

Comment (marcellorinaldo) [reply kura/org.eclipse.kura.cloudconnection.sparkplug.mqtt.provider/src/main/java/org/eclipse/kura/cloudconnection/sparkplug/mqtt/message/SparkplugBProtobufPayloadBuilder.java]: Its a lefover, will remove

**Review by pierantoniomerlino — APPROVED** [review]

**Comment by marcellorinaldo** [comment]
Licensing should be in order. The used protobuf definition file (`.proto`) is licensed under EPL 2.0 and is not modified in this contribution.

Same principle has been applied to PR https://github.com/eclipse/kura/pull/3810, where the contribution included a protobuf definition from a third party.

**Review by MMaiero — APPROVED** [review]

**Pull request merged**
